### ISSUE TITLE
Have you encountered label leakage in the inference time?

### ISSUE THREAD

**Issue by A11en0** [Networks-Learning/improve-expert-predictions-conformal-prediction#1]
Hi, thanks for your wonderful job! When I review code in line 148 of conformal_prediction.py, I find that the test label is used to calculate the confusion matrix, I think it if encountered data leakage in the inference time. Could you help me to answer the question. Thanks again! 

<img width="1163" alt="image" src="https://github.com/Networks-Learning/improve-expert-predictions-conformal-prediction/assets/33925232/163e50bb-7dbc-470d-b4bf-1eb3800c754d">

**Comment by ElenStr** [comment]
We are excited to hear that you appreciate our work! We would like to clarify that in the highlighted lines we do **not** use the test label to calculate the confusion matrix. We use the test label to select the entries of the confusion matrix that we need to compute the MNL model for each test sample and prediction set, so that we can simulate the performance of human experts.  The MNL model essentially captures the distribution of the human predictions given a prediction set in the context of the ground truth label (without the ground truth label being revealed to the human). To compute the MNL model, one needs access to the entries of the confusion matrix of the human predictions alone given the ground truth label. In our latest work [Designing Decision Support Systems Using Counterfactual Prediction Sets](https://arxiv.org/abs/2306.03928)  we performed a large scale real data evaluation, where we collected real human predictions given prediction sets; code and data are available in https://github.com/Networks-Learning/counterfactual-prediction-sets."

**Comment by A11en0** [comment]
Fine, thanks. But I am also have a little trouble with the equation (8)

<img width="524" alt="image" src="https://github.com/Networks-Learning/improve-expert-predictions-conformal-prediction/assets/33925232/a8844bf0-4b47-47c0-b61d-cd5ced5877d4">

I read the original paper of Kerrigan et al. (2021), the inference function combines the model's output logits and confusion matrix, but why in your paper only contain the confusion matrix?
